src/components/Integrations/Shopify/Shopify.js

```diff
@@ -99,37 +86,61 @@ const Shopify = ({ intl, dependencies: { shopifyClient } }) => {
     </>
   );
 
+  useInterval({
+    runOnStart: true,
+    delay: 20000,
+    callback: async () => {
+      const result = await shopifyClient.getShopifyData();
+      if (!result.success) {
+        setShopifyState({
+          error: <FormattedHTMLMessage id="validation_messages.error_unexpected_HTML" />,
+          isLoading: false,
+        });
+      } else if (result.value.length) {
+        setShopifyState({
+          error: null,
+          shops: result.value,
+          isConnected: true,
+          isConnecting: false,
+          isLoading: false,
+        });
+      } else {
+        setShopifyState({ error: null, shops: result.value, isConnected: false, isLoading: false });
```

Comment:
```suggestion
        setShopifyState({ shops: [] });
```

Reply:
`IsLoading: false`, I need it to make the spinner dissapear, the same in the previous case.

Reply:
We have talked, the confusion is that @cbernat assumes that this works as class components _setState_ function, and I assume that it simply updates a value.

She will try and then let us know.

Reply:
@andresmoschini It works without merging state. So I'll keep the change done. Thanks!